reactome_gnn/train.py

```diff
@@ -0,0 +1,143 @@
+import copy
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+import torch.nn.functional as F
+import torch.optim as optim
+from dgl.dataloading import GraphDataLoader
+
+from reactome_gnn import model, dataset, hyperparameters
+
+
+def draw_loss_plot(train_loss, valid_loss, save_path):
+    """Draw and save plot of train and validation loss over epochs.
+
+    Parameters
+    ----------
+    train_loss : list
+        List of training loss for each epoch
+    valid_loss : list
+        List of validation loss for each epoch
+    name : str
+        A name used for the file
+
+    Returns
+    -------
+    None
+    """
+    plt.figure()
+    plt.plot(train_loss, label='train')
+    plt.plot(valid_loss, label='validation')
+    plt.title('Loss over epochs')
+    plt.xlabel('Epoch')
+    plt.ylabel('Loss')
+    plt.legend()
+    plt.savefig(f'{save_path}.png')
+    plt.show()
+
+
+def train(hyperparams=None, data_path='demo/data/example', plot=True):
+    """Train a network which will produce the embeddings.
+
+    Parameters
+    ----------
+    hyperparams : dict, optional
+        Dictionary with hyperparameters
+    data_path : str, optional
+        Relative path to where the data is stored
+    plot : bool, optional
+        Whether to plot loss over epochs
+
+    Returns
+    -------
+    None
+    """
+    if hyperparams is None:
+        hyperparams = hyperparameters.get_hyperparameters()
+
+    # Unpack hyperparameters
+    num_epochs = hyperparams['num_epochs']
+    dim_latent = hyperparams['dim_latent']
+    num_layers = hyperparams['num_layers']
+    learning_rate = hyperparams['lr']
+    batch_size = hyperparams['batch_size']
+    device = hyperparams['device']
+
+    model_path = os.path.join(data_path, 'models')
+    model_path = os.path.join(model_path, f'model_dim{dim_latent}_lay{num_layers}.pth')
+    
+    # Create datasets
+    ds = dataset.PathwayDataset(data_path)
+    ds_train, ds_valid = [ds[0], ds[1], ds[2]], [ds[3], ds[4]]
```

Comment:
It's interesting that the train and validation have the same curve even with this line of code.

Reply:
Yes, this is a bit "hardcoded", but simply because there is not a lot of graphs in this toy dataset, which is also the reason for the same curves. With more graphs in the dataset we could simply split the dataset randomly into train/validation, and with more graphs in both train/validation dataset I'm sure the losses would be more noisy. I could try to do that in the next few days.